res/controllers/Reloop-Digital-Jockey-2-IE.scripts.js

```diff
@@ -0,0 +1,876 @@
+        this.outKey = "button_parameter3";
+        this.connect();
+        this.trigger();
+    },
+    shift: function() {
+        this.disconnect();
+        this.group = "[QuickEffectRack1_[Channel" + this.channelNr + "]_Effect1]";
+        this.inKey = "enabled";
+        this.outKey = "enabled";
+        this.connect();
+        this.trigger();
+    },
+});
+
+
+////////////////////////////////////////////////////////////////////////
+// Library                                                            //
+////////////////////////////////////////////////////////////////////////
+
+/* Trax knob */
+
+RDJ2.TraxKnob = function(options) {
+    components.Encoder.call(this, options);
+};
+RDJ2.TraxKnob.prototype = new components.Encoder({
+    group: "[Library]",
+    unshift: function() {
+        this.inKey = "MoveVertical";
+    },
+    shift: function() {
+        this.inKey = "ScrollVertical";
+    },
+    input: function(channel, control, value) {
+        var knobDelta = RDJ2.getKnobDelta(value);
+        this.inSetValue(knobDelta);
+    }
+});
+
+/* Trax button */
+
+RDJ2.TraxButton = function(obj) {
+    this.detectedDecks = [];
+    /* group and/or outKey cannot be defined at prototype initialization
+       (inside anonymous object passed to components.Button constructor
+       below) as this would cause additional premature engine.makeConnection
+       call that binds prototype object's output callback to the outKey.
+       This would cause uncaught exception when invoking the callback:
+
+       "TypeError: Result of expression 'this.detectedDecks' [undefined] is not an object"
+
+       which happens because at the time we create the prototype object,
+       we have no information yet about detectedDecks, as detectedDecks is
+       created at new TraxButton object construction, when its prototype is
+       long time existing. */
+    this.group = "[Master]";
+    this.outKey = "maximize_library";
+
+    this.detectDecks(obj);
+    components.Button.call(this);
+};
+RDJ2.TraxButton.prototype = new components.Button({
+    unshift: function() {
+        this.type = components.Button.prototype.types.toggle;
+        this.group = "[Master]";
+        this.inKey = "maximize_library";
+    },
+    shift: function() {
+        this.type = components.Button.prototype.types.push;
+        this.group = "[Library]";
+        this.inKey = "MoveFocusForward";
+    },
+    output: function(value) {
+        this.updateTraxMode(value);
+    },
+    updateTraxMode: function(mode) {
+        this.detectedDecks.forEach(function(deck) {
+            deck.jogModeSelector.setTraxMode(mode);
+        });
+    },
+    detectDecks: function(obj) {
+        // find decks in the passed object and store them in the array
+        for (var memberName in obj) {
+            // eslint-disable-next-line no-prototype-builtins -- TODO analyze the cause
+            if (obj.hasOwnProperty(memberName) && obj[memberName] instanceof components.Deck) {
```

Comment:
```suggestion
            if (Object.prototype.hasOwnProperty.call(obj, memberName) && obj[memberName] instanceof components.Deck) {
```
ESLint warns that `obj`, might not inherit from `Object`, so `hasOwnProperty` could be undefined on obj. See https://eslint.org/docs/rules/no-prototype-builtins

Reply:
Thanks for the hint!
Done in the latest commit.